src/Umbraco.Core/Sync/BatchedDatabaseServerMessenger.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using Umbraco.Core.Models.Rdbms;
+using umbraco.interfaces;
+
+namespace Umbraco.Core.Sync
+{
+    // abstract because it needs to be inherited by a class that will
+    // - trigger FlushBatch() when appropriate
+    // - trigger Boot() when appropriate
+    // - trigger Sync() when appropriate
+    //
+    public abstract class BatchedDatabaseServerMessenger : DatabaseServerMessenger
+    {
+        private readonly Func<bool, ICollection<RefreshInstructionEnvelope>> _getBatch;
+
+        protected BatchedDatabaseServerMessenger(ApplicationContext appContext, bool enableDistCalls, DatabaseServerMessengerOptions options,
+            Func<bool, ICollection<RefreshInstructionEnvelope>> getBatch)
```

Comment:
Wouldn't it be easier instead of having a delegate in the ctor to just have an abstract method with the same signature?